spec/requests/api/v2/volume_update_spec.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+require './spec/requests/api/v2/shared_examples/api_key_authenticatable'
+
+describe 'VolumeUpdate API', with: :api_v2 do
+  let(:base_endpoint) { '/api/v2/volume_updates' }
+
+  it_behaves_like 'ApiKeyAuthenticatable'
+
+  context 'with multiple VolumeUpdates' do
+    before { create_list(:volume_update, 5) }
+
+    it 'returns a list of volume updates' do
+      api_get base_endpoint
+
+      expect(response).to have_http_status(:success)
+      expect(json['data'].length).to eq(5)
+    end
+  end
+
+  context 'with a VolumeUpdate' do
+    let(:resource_model) { create(:volume_update) }
+    let(:updated_resource_model) do
+      resource_model.update(volume_change: 10.0)
+      resource_model
+    end
+
+    let(:payload) do
+      {
+        'data' => {
+          'id' => resource_model.id,
```

Comment:
I'm not super comfortable with the way `resource_model` and `updated_resource_model` return the same object.  The second one has side-effects on the first one so it's possble you refer to `resource_model` without realising that it too has the updated `volume_change` value.  I think (by luck and lazy loading of `let` constructs) that because you haven't yet referred to the updated_resource_model in this block, it actually hasn't been modified, but this is a bit of a code smell for me.  I think it would be better to have a single `resource_model` variable and have a method you call when you want to have it updated.  Something like `def update_volume_change(model) { model.update(volume_change: 10.0) }`

Reply:
Yes much better!